chromium_src/ui/views/controls/button/md_text_button.h

```diff
@@ -41,9 +70,31 @@ class VIEWS_EXPORT MdTextButton : public MdTextButtonBase {
 
   SkPath GetHighlightPath() const;
 
+  Kind GetKind() const;
+  void SetKind(Kind kind);
+
+  void SetIcon(const gfx::VectorIcon* icon);
+
+  bool GetLoading() const;
+  void SetLoading(bool loading);
+
+  // Until we decide to update the whole UI to use the new Leo colors, we
+  // need to keep this logic around. Currently the new colors are opt-in only.
+  void UpdateOldColorsForBrave();
+
+  // MdTextButtonBase:
+  void UpdateColorsForBrave() override;
+  void UpdateIconForBrave() override;
+
  protected:
   // views::Views
   void OnPaintBackground(gfx::Canvas* canvas) override;
+
+ private:
+  Kind kind_ = OLD;
+  bool loading_ = false;
+  absl::optional<ButtonTheme> theme_;
```

Comment:
b1. continued from [a](https://github.com/brave/brave-core/pull/15149#discussion_r983015225), using map could let us have cleaner code.

This idea is similar to how upstream folks manage icons or color per button states. https://source.chromium.org/chromium/chromium/src/+/main:ui/views/controls/button/label_button.h;l=228?q=label_button&ss=chromium

```suggestion
  base::flat_map<ButtonState, ButtonStyle> style_per_state_;
```

Reply:
I'm using your idea with the `StyleKey`